src/classes/SftpSessionHandler.ts

```diff
@@ -1070,6 +1070,12 @@ export class SftpSessionHandler {
   }
 
   private getCurrentPermanentFileSystem(): PermanentFileSystem {
+    if (
+      this.authenticationSession.tokenExpired()
+      || this.authenticationSession.tokenWouldExpireSoon()
```

Comment:
This is not where I'd expect this check to live, as it is not obvious to me that it will be made before each and every API request. I'm not sure where it would better be placed; perhaps in `getClientConfiguration`? It would be nice if the Permanent SDK helped us out here. @slifty what do you think?

Reply:
Agreed @jasonaowen!  This check should be done *when accessing the authToken* -- I expand on this concept in more detail in another comment on this review.